### PR TITLE
Update Dependabot configuration for major updates only

### PR DESCRIPTION
**Pull request by BenTaylor-TfN**
Configure Dependabot to allow only major version updates and set a limit on open pull requests.

# Tasks

- [ ] Have unittests been added (testing individual functions and their edge cases)
- [ ] Has documentation (including user guide) been updated
- [ ] Have new dependencies been added (or changed) in relevant `requirements.txt`
- [ ] Code linting, tests and other workflows pass

### CODE REVIEW

File: {{ cookiecutter.project_slug }}/.github/dependabot.yml [hunk 1]
```diff
@@ -9,6 +9,10 @@ updates:
...
     schedule:
       interval: "weekly"
-    # This will tell dependabot to only submit security update PRs
-    # If you want to bump minimum versions when available, increase this number. We recommend a maximum value of 3.
-    open-pull-requests-limit: 0
+    # Only allow major version updates to reduce noise from minor/patch updates
+    allow:
+      - dependency-name: "*"
+        update-types:
+          - "version-update:semver-major"
```
Comment (MattBuckley-TfN) [inline {{ cookiecutter.project_slug }}/.github/dependabot.yml]: Allow minor versions for caf packages because:
- Lots of caf packages are in 0.x stage so major updates are hidden until 1.0 release
- Remind people of a new caf release, basically advertise caf releases directly on other repos
- Makes it easier for us to see if a new version of one caf package breaks a different caf package
```suggestion

    allow:
      # Only allow major version updates to reduce noise from minor/patch updates
      - dependency-name: "*"
        update-types:
          - "version-update:semver-major"

      # Allow minor versions for caf packages to ensure new versions are tested more often
      - dependency-name: "caf.*"
        update-types:
          - "version-update:semver-minor"

```

**Review by MattBuckley-TfN — CHANGES_REQUESTED** [review]
Added minor versions for caf packages.

**Review by MattBuckley-TfN — APPROVED** [review]
LGTM
